spec/lib/mcb/courses_editor_spec.rb

```diff
@@ -258,6 +258,22 @@ def run_editor(*input_cmds)
       end
     end
 
+    context 'when there are several courses with the same course code' do
+      let(:another_provider) { create(:provider) }
+      let!(:another_course_with_the_same_course_code) {
+        create(:course,
+               provider: another_provider,
+               course_code: course.course_code,
+               name: 'Another name')
+      }
+
+      it 'edits the course from the specified provider' do
+        expect { run_editor("edit title", "Mathematics", "exit") }.
+          to change { course.reload.name }.
+          from("Original name").to("Mathematics")
+      end
+    end
+
     describe 'tries to edit a non-existent course' do
       let(:course_codes) { [course_code, "ABCD"] }
```

Comment:
could extend this to test in case where `ABCD` does exist with another provider

Reply:
added and force-pushed